app/src/main/java/br/com/mgalhardo/guidebook/dagger/UiComponent.java

```diff
@@ -0,0 +1,12 @@
+package br.com.mgalhardo.guidebook.dagger;
+
+import br.com.mgalhardo.guidebook.dagger.module.presentation.PresenterModule;
+import br.com.mgalhardo.guidebook.dagger.scope.PerActivity;
+import br.com.mgalhardo.guidebook.presentation.ui.guidelist.GuideListActivity;
+import dagger.Subcomponent;
+
+@PerActivity
```

Comment:
@viniciussoares Sorry if I couldn't explain it (it is a beginner's question). I understand the presenters should use the  @PerActivity scope. My question is why the UiComponent also uses this scope. I aggree the presenters should be destroyed together with the activities to avoid leaks. My question was regarding the Component. Should a new instance (of the UiComponent) be created for every activity you inject it into? Thanks! =)